### PR TITLE
Space-Travelers(Group Project)

### PR DESCRIPTION
**Pull request by edubew**
### 1. Config & Basic setup
- [x] Create React App.
- [x] Install [React Redux](https://react-redux.js.org/), [Redux Logger](https://www.npmjs.com/package/redux-logger) and [React Router](https://reactrouter.com/web/guides/quick-start).
- [x] Create routes and view components: rockets, missions, my profile.
- [x] Use `<NavLink />` for the page navigation links and style active class to indicate which section/page the user is currently on (underline active navigation link).

### 2. Redux: Fetch data and update Redux store
- [x] Upon first render fetch data from the SpaceX API endpoints:
  - [Rockets:](url) https://api.spacexdata.com/v4/rockets
  - [Missions:](url) https://api.spacexdata.com/v3/missions
- [x] Once the data are fetched, dispatch an action to store the selected data in Redux store
- [x] Use useSelector() Redux Hook to select the state slices and render lists of rockets and missions in corresponding routes
- [x] Write actions and reducers for booking rockets and joining missions
- [x] Write actions and reducers for canceling rockets and leaving missions

### 3. Render UI: conditional components rendering
- [x] Rockets that have already been reserved should show a "Reserved" badge and "Cancel reservation" button instead of the default "Reserve rocket"
- [x] Missions that the user has joined already should show a badge "Active Member" instead of the default "NOT A MEMBER" and a button "Leave Mission" instead of the "Join Mission" button

### 4. Render UI: My Profile section
- [x] Render a list of all joined missions (use filter()).
- [x] Render a list of all reserved rockets (use filter()).

### CODE REVIEW

File: src/Redux/rockets/rocketsSlice.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+const url = 'https://api.spacexdata.com/v4/rockets';
+
+const FETCH_ROCKETS = 'space-travelers-react-redux/rockets/fetchRockets';
+
+// Action creator
+export const fetchRockets = createAsyncThunk(FETCH_ROCKETS, async () => {
+  const response = await fetch(url);
+  const data = await response.json();
+  return data;
+});
```
Comment (binhussen) [inline src/Redux/rockets/rocketsSlice.js]: - Kindly, you are storing the whole API response object which is not necessary, Please map and store only used data to redux store to improve its performance as you did for missions👍

File: src/Components/Rocket.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { makeReservation, cancelReservation } from '../Redux/rockets/rocketsSlice';
+
+const Rocket = ({
+  name, description, image, id, reserved,
+}) => {
+  const dispatch = useDispatch();
```
Comment (binhussen) [inline src/Components/Rocket.js]: - Kindly, you are dispatching the state when you navigate to the page which is resetting(clearing the reserved rockets) the state, please dispatch the action for the first time only using `useEffect` as you did for missions 👍 to have the correct data.
- Kindly use also `useSelector` for getting the updated state from the store as you did in missions 👌

**Review by binhussen — CHANGES_REQUESTED** [review]
Hi @edubew and @QueenterJuma , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- All linter checks passed ✔️
-Used React-redux and testing library ✔️
- Used Git flow ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Redux/rockets/rocketsSlice.js (hunk 1 above)
Comment (QueenterJuma) [reply src/Redux/rockets/rocketsSlice.js]: Resolved

File: src/Components/Rocket.js (hunk 2 above)
Comment (QueenterJuma) [reply src/Components/Rocket.js]: Resolved: using useSelector and maintaining state after the first API fetch

**Review by OyePriscilla — APPROVED** [review]
Hi @edubew and @QueenterJuma 👏 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Use React components ✔️
Use React props ✔️
Use React Router ✔️
Connect React and Redux ✔️
Implement conditional components rendering ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**